bftengine/src/bcstatetransfer/SourceSelector.hpp

```diff
@@ -134,6 +144,17 @@ class SourceSelector {
   uint16_t nominatedPrimaryCounter_ = 0;
   uint16_t minPrePrepareMsgsForPrimaryAwarness_ = 10;
   logging::Logger &logger_;
+
+ public:
+  // Metric
+  concordMetrics::Component metrics_component_;
+  struct Metrics {
+    StatusHandle preferred_replicas_;
```

Comment:
You missed 3 reasons for replacement. Here are all the reasons currently for replacment:

1. no source
2. bad data
3. too many retransmissions
4. periodic
5. source is primary

I suggest prefixing all metrics with same string with "replacment_due_" ...

Reply:
Thanks. Addressed.